spec/factories/games.rb

```diff
@@ -13,5 +13,50 @@
       white_player_id 1
       black_player_id 2
     end
+
+    trait :check_scenario do
```

Comment:
We should likely have tests that check a knight-puts-king-into check scenario as it's different.

Good to check x-attack, y-attack and diagonal attacks as well